src/Stack/Setup/Installed.hs

```diff
@@ -39,17 +39,22 @@ import           RIO.Process
 data Tool
     = Tool PackageIdentifier -- ^ e.g. ghc-7.8.4, msys2-20150512
     | ToolGhcjs ActualCompiler -- ^ e.g. ghcjs-0.1.0_ghc-7.10.2
+    | ToolGhcGit !Text !Text   -- ^ e.g. ghc-git-COMMIT_ID-FLAVOUR
```

Comment:
Going in the direction of allowing Git-build GHC's to be used on platforms that don't have the relevant bindists: what would you think about including the `Version` of the built compiler here? Then we could theoretically have a setting in `stack.yaml` which says "try to use any Git-built GHC if it's available".

Reply:
> what would you think about including the Version of the built compiler here? 
> Then we could theoretically have a setting in stack.yaml which says "try to use any Git-built GHC if it's available".

I would prefer if we start dealing with reported version in another PR (in the context of #4637 maybe). I like the fact that a ghc-git "version" is almost fully determined by commitID+flavour (it's not nix so it doesn't determine external tools such as GCC but we're close as we even determine the bootstrap compiler). If we start to infer a SemVer version from it, then maybe we will have to infer the integer-library used, the configure flags (e.g. RELEASE=YES/NO), etc. and try to match all this against stack passed options (ghc-variant, ghc-build, compiler version...).

I really think it deserves more thoughts and its own PR ;)

Reply:
Fair enough, you make a good point 👍